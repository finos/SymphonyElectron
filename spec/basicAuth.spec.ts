import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import BasicAuth from '../src/renderer/components/basic-auth';

describe('basic auth', () => {
    const usernameTargetMock: object = { target: { id: 'username', value: 'foo' }};
    const passwordTargetMock: object = { target: { id: 'password', value: '123456' }};
    const basicAuthMock: object = {
        hostname: 'example',
        isValidCredentials: true,
        password: '123456',
        username: 'foo',
    };
    const usernameMock = { username: 'foo' };
    const passwordMock = { password: '123456' };

    it('should render correctly', () => {
        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
        expect(wrapper).toMatchSnapshot();
    });

    it('should call `setState` when `change` is called', () => {
        const spy: jest.SpyInstance = jest.spyOn(BasicAuth.prototype, 'setState');
        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
        wrapper.find('#username').simulate('change', usernameTargetMock);
        expect(spy).lastCalledWith(usernameMock);
        wrapper.find('#password').simulate('change', passwordTargetMock);
        expect(spy).lastCalledWith(passwordMock);
    });

    it('should call submit login', () => {
        const { ipcRenderer } = require('./__mocks__/electron');
        const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, 'send');
        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
        wrapper.find('#username').simulate('change', usernameTargetMock);
        wrapper.find('#password').simulate('change', passwordTargetMock);
        wrapper.find('#basicAuth').simulate('submit');
        expect(spy).lastCalledWith('basic-auth-login', { ...usernameMock, ...passwordMock });
    });

    it('should call `basic-auth-closed` event when cancel button is clicked', () => {
        const { ipcRenderer } = require('./__mocks__/electron');
        const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, 'send');
        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
        wrapper.find('#cancel').simulate('click');
        expect(spy).lastCalledWith('basic-auth-closed', false);
    });

    describe('basic auth mount and unmount event', () => {
        const { ipcRenderer } = require('./__mocks__/electron');
        const basicAuthDataLabel: string = 'basic-auth-data';
        const onLabelEvent: string = 'on';
        const removeListenerLabelEvent: string = 'removeListener';

        it('should call `basic-auth-data` event when component is mounted', () => {
            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, onLabelEvent);
            shallow(React.createElement(BasicAuth));
            expect(spy).toBeCalledWith(basicAuthDataLabel, expect.any(Function));
        });

        it('should remove listen `basic-auth-data` when component is unmounted', () => {
            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, removeListenerLabelEvent);
            const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
            wrapper.unmount();
            expect(spy).toBeCalledWith(basicAuthDataLabel, expect.any(Function));
        });

        it('should call `updateState` when component is mounted', () => {
            const spy: jest.SpyInstance = jest.spyOn(BasicAuth.prototype, 'setState');
            shallow(React.createElement(BasicAuth));
            ipcRenderer.send('basic-auth-data', basicAuthMock);
            expect(spy).toBeCalledWith(basicAuthMock);
        });
    });
});
