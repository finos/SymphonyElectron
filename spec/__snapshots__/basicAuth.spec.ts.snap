// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic auth should render correctly 1`] = `
<div
  className="container"
>
  <span>
    Please provide your login credentials for:
  </span>
  <span
    className="hostname"
  >
    unknown
  </span>
  <span
    className="credentials-error"
    id="credentialsError"
  >
    Invalid user name/password
  </span>
  <form
    action="Login"
    id="basicAuth"
    name="Basic Auth"
    onSubmit={[Function]}
  >
    <table
      className="form"
    >
      <tbody>
        <tr>
          <td
            id="username-text"
          >
            User name:
          </td>
          <td>
            <input
              id="username"
              name="username"
              onChange={[Function]}
              required={true}
              title="Username"
            />
          </td>
        </tr>
        <tr>
          <td
            id="password-text"
          >
            Password:
          </td>
          <td>
            <input
              id="password"
              name="password"
              onChange={[Function]}
              required={true}
              title="Password"
              type="password"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="footer"
    >
      <div
        className="button-container"
      >
        <button
          id="login"
          type="submit"
        >
          Log In
        </button>
      </div>
      <div
        className="button-container"
      >
        <button
          id="cancel"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
</div>
`;
